pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -517,53 +517,71 @@ func (vs *VSphere) InstanceExistsByProviderID(providerID string) (bool, error) {
 
 // InstanceID returns the cloud provider ID of the node with the specified Name.
 func (vs *VSphere) InstanceID(nodeName k8stypes.NodeName) (string, error) {
```

Comment:
Can you please verify the steps on how did you test this scenario.

Reply:
Launced k8s cluster and then destroyed VM. Before fix: k8s was reporting node as not ready. After fix: node object representing that VM in k8s was deleted.